packages/resource/src/index.ts

```diff
@@ -0,0 +1,237 @@
+      return response;
+    },
+    (s: S) => {
+      delete localCache[serializer(s)];
+      save();
+    },
+    expired,
+  ];
+}
+
+export type RetryOptions = {
+  delay?: number;
+  retries?: number;
+};
+
+/**
+ * **Creates a fetcher that retries multiple times in case of errors**
+ * ```ts
+ * const fetcher = makeRetrying((url) => fetch(url), { retries: 5 });
+ * ```
+ * Receives the fetcher and an optional options object and returns a wrapped fetcher that retries on error after a delay multiple times.
+ *
+ * The optional options object contains the following optional properties:
+ * - `delay` - number of Milliseconds to wait before retrying; default is 5s
+ * - `retries` - number of times a request should be repeated before giving up throwing the last error; default is 3 times
+ */
+export function makeRetrying<S, T, R = unknown>(
+  fetcher: ResourceFetcher<S, T, R>,
+  options: RetryOptions = {},
+): ResourceFetcher<S, T, R> {
+  const delay = options.delay ?? 5000;
+  let retries = options.retries || 3;
+  const retrying: ResourceFetcher<S, T, R> = async (k: S, info: ResourceFetcherInfo<T, R>) => {
+    try {
+      return await fetcher(k, info);
+    } catch (error: unknown) {
+      if (retries-- > 0) {
+        delay && await new Promise(resolve => setTimeout(() => resolve, delay));
+        return retrying(k, info);
+      } else {
+        retries = options.retries || 3;
+        throw error;
+      }
+    }
+  };
+  return retrying;
+}
+
+function toArray(item: any) {
+  return Array.isArray(item) ? item : item ? [item] : [];
+}
+
+/**
+ * **Automatically aggregates resource changes**
+ * ```ts
+ * const pages = makeAggregated(currentPage, []);
+ * ```
+ * Depending on the content of the initialValue or the first response, this will aggregate the incoming responses:
+ * - null will not overwrite undefined
+ * - if the previous value is an Array, incoming values will be appended
+ * - if any of the values are Objects, the current one will be shallow-merged into the previous one
+ * - if the previous value is a string, more string data will be appended
+ * - otherwise the incoming data will be put into an array
+ *
+ * Objects and Arrays are re-created on each operation, but the values will be left untouched, so `<For>` should work fine.
+ */
+export function makeAggregated<R, I extends R | R[]>(res: Resource<R>, initialValue?: I) {
+  return createMemo<I | R | {} | R[] | I[] | undefined>(previous => {
+    const current = res();
+    return current == null && previous == null
+      ? previous
+      : Array.isArray(previous || current)
+      ? [...toArray(previous), ...toArray(current)]
+      : typeof (previous || current) === "object"
+      ? Object.assign({}, previous || {}, current || {})
+      : typeof previous === "string" || typeof current === "string"
+      ? (previous?.toString() || "") + (current || "")
+      : previous
+      ? [previous, current]
+      : [current];
+  }, initialValue);
```

Comment:
I feel like this could be also exported as a separate helper.
Like `aggregate` or `getAggregated` just as a vanilla js pure function.